### PR TITLE
add section for debugging applications under test

### PR DESCRIPTION
**Pull request by EndangeredMassa**
Thanks to @toolbear for figuring out these steps. Source: https://github.com/groupon-testium/testium/issues/141

Can a few people test this out and suggest any improvements to the list of steps?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -219,6 +219,22 @@ to save the commands you entered into a file.
...
+
+1. run [`node-inspector`](https://github.com/node-inspector/node-inspector)
+1. load the URL from the output in Google Chrome
+1. set `launch = false` in `.testiumrc`
+1. specify constant port number by setting `app.port` in `.testiumrc` OR `export PORT=something`
+1. add `debugger` statement in production code
+1. run your app with `node --debug-brk app.js` (or the equivalent of what `npm start` does for you with `--debug-brk` argument passed to node)
+1. run tests as normal (not under debugger)
+
+Gotchas: don't simultaneously run your app under the debugger
```
Comment (jkrems) [inline README.md]: Maybe one more gotcha: if you use `cluster`, you might want to start the worker process directly.

File: README.md [hunk 2]
```diff
@@ -219,6 +219,22 @@ to save the commands you entered into a file.
 For more info,
 read the [official node repl docs](http://nodejs.org/api/repl.html).
 
+
+## Debugging Application Under Test
+
+1. run [`node-inspector`](https://github.com/node-inspector/node-inspector)
+1. load the URL from the output in Google Chrome
```
Comment (khoomeister) [inline README.md]: The newest `node-inspector` opens the URL in your default browser automatically.

Comment (khoomeister) [inline README.md]: Whoops sorry, I was thinking about `node-debug`.

Comment (jkrems) [inline README.md]: Good point - doesn't `node-debug` ship with `node-inspector`? Looks like the even advertise it as the default way to start node-inspector: https://github.com/node-inspector/node-inspector#start

File: README.md (hunk 1 above)
Comment (EndangeredMassa) [inline README.md]: Good point!

**Comment by EndangeredMassa** [comment]
Updated.

**Comment by jkrems** [comment]
:+1:

**Comment by toolbear** [comment]
Possibly one step simpler:

1. set `launch = false` in `.testiumrc`
2. specify constant port number by setting `app.port` in `.testiumrc` *or* `export PORT=something`
3. add `debugger` statement in production code
4. run your app with `node-debug app.js` (or the equivalent of what npm start does for you, replacing `node` with `node-debug`)
5. run tests as normal (not under debugger)

The `node-debug` usage is decent enough to lead someone to `--debug-brk` with minimal digging.

**Comment by toolbear** [comment]
Previous "one step simpler" comment doesn't work in practice.

**Comment by toolbear** [comment]
Had cause to debug my app again. Followed the current steps as committed here and they worked for me.

:+1: for merging

**Comment by jkrems** [comment]
@toolbear Thanks for the confirmation! Still good to have the reassurance that the instructions work :)

**Pull request merged**